src/firmware/app/primitives/primitive_manager.c

```diff
@@ -3,8 +3,13 @@
 // There are different semaphore implementations depending on if we're on a x86 or arm
 // system, and we use typdefs here to switch between them
 #ifdef __arm__
-#include <FreeRTOS.h>
+
+// clang-format off
```

Comment:
could we add a comment about why clang-format is off here? also should it just wrap `FreeRTOS.h` and not `semphr.h`?

Reply:
added a comment

Reply:
Hmm, could we expand on this a bit? Why do we need `FreeRTOS.h` to be first?

Reply:
added a comment, it basically doesn't compile with
``` #error "include FreeRTOS.h" must appear in source files before "include semphr.h" ```